WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginMagicLinkRequestFragment.java

```diff
@@ -0,0 +1,239 @@
+        super.onAttach(context);
+        if (context instanceof LoginListener) {
+            mLoginListener = (LoginListener) context;
+        } else {
+            throw new RuntimeException(context.toString() + " must implement LoginListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mLoginListener = null;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        outState.putBoolean(KEY_IN_PROGRESS, mInProgress);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.menu_login, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.help) {
+            if (mLoginListener != null) {
+                mLoginListener.help();
+            }
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mDispatcher.register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        mDispatcher.unregister(this);
+    }
+
+    private void showMagiclinkRequestProgressDialog() {
+        startProgress(getString(R.string.login_magic_link_email_requesting));
+    }
+
+    protected void startProgress(String message) {
+        mRequestMagicLinkButton.setEnabled(false);
+        mProgressDialog = ProgressDialog.show(getActivity(), "", message, true, true,
+                new DialogInterface.OnCancelListener() {
+                    @Override
+                    public void onCancel(DialogInterface dialog) {
+                        if (mInProgress) {
+                            endProgress();
+                        }
+                    }
+                });
+        mInProgress = true;
+    }
+
+    protected void endProgress() {
+        mInProgress = false;
+
+        if (mProgressDialog != null) {
+            mProgressDialog.cancel();
+            mProgressDialog = null;
+        }
+
+        // nullify the reference to denote there is no operation in progress
+        mProgressDialog = null;
+
+        mRequestMagicLinkButton.setEnabled(true);
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAuthEmailSent(AccountStore.OnAuthEmailSent event) {
+        if (!mInProgress) {
+            // ignore the response if the magic link request is no longer pending
+            return;
+        }
+
+        endProgress();
+
+        if (event.isError()) {
+            AppLog.e(AppLog.T.API, "OnAuthEmailSent has error: " + event.error.type + " - " + event.error.message);
+            if (isAdded()) {
+                ToastUtils.showToast(getActivity(), R.string.magic_link_unavailable_error_message,
+                        ToastUtils.Duration.LONG);
+            }
+        } else {
```

Comment:
I didn't like that if/else nesting anyway so I "un-nested" it with c8f08a9. Thanks!